db/merge_helper.cc

```diff
@@ -153,12 +154,12 @@ Status MergeHelper::MergeUntil(InternalIterator* iter,
     ParsedInternalKey ikey;
     assert(keys_.size() == merge_context_.GetNumOperands());
 
-    if (ParseInternalKey(iter->key(), &ikey) != Status::OK()) {
+    Status pik_status =
+        ParseInternalKey(iter->key(), &ikey, allow_data_in_errors);
+    if (!pik_status.ok()) {
       // stop at corrupted key
       if (assert_valid_internal_key_) {
```

Comment:
[existing issue] It looks like this is initialized to `false` (https://github.com/facebook/rocksdb/blob/a16d1b2fd36a2fe036b0aed1008f60570c981773/db/compaction/compaction_job.cc#L897). Is it a bug? In any case I think we should get rid of `MergeHelper:: assert_valid_internal_key_` and also `CompactionIterator::expect_valid_internal_key_` -- they seem error prone and not useful since we always want key corruptions to be reported.

Reply:
Agree. I will fix it in a separate PR.